packages/grid/x-data-grid/src/internals/components/base/GridBody.tsx

```diff
@@ -72,7 +73,9 @@ function GridBody(props: GridBodyProps) {
 
   return (
     <GridMainContainer>
-      <GridOverlays />
+      <GridOverlaysRoot>
+        <GridOverlays />
```

Comment:
Could `GridOverlaysRoot` be the root element of `GridOverlays`? This approach kinda breaks the internal convention for styled components:

```tsx
<GridOverlays>
  <GridOverlaysRoot>
    {currentOverlay}
  </GridOverlaysRoot>
</GridOverlays>
```

For the error overlay we can make an exception and export `GridOverlaysRoot`.

Reply:
Agree!

Reply:
I've kept `GridOverlayWrapper` in separate file though to avoid exporting it in the package